### PR TITLE
feature-8791: Consent of refund policy for the attendee form

### LINKED ISSUE #8791: Consent of refund policy for the attendee form
Please add a consent option for the refund policy.

* I agree to the terms of the refund policy of the event.

Ensure that all text is translatable

### PR DESCRIPTION
**Pull request by lthanhhieu**
Fixes #8791

#### Short description of what this resolves:
- Consent of refund policy for the attendee form

#### Changes proposed in this pull request:
- Consent of refund policy for the attendee form

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Comment by mariobehling** [comment]
Rate limit of CI. LGTM.

Please always add screenshot in future.

File: app/mixins/custom-form.js [hunk 1]
```diff
@@ -661,12 +661,23 @@ export default Mixin.create(MutableArray, {
...
+        isPublic        : false,
+        event           : parent
+      }),
       this.store.createRecord('custom-form', {
         name            : 'Consent form field',
         fieldIdentifier : 'is_consent_form_field',
         formID          : _formID,
         form            : 'attendee',
-        type            : 'select',
+        type            : 'checkbox',
```
Comment (norbusan) [inline app/mixins/custom-form.js:680]: Does this change make other changes in the table format necessary?

File: app/components/forms/orders/order-form.js [hunk 2]
```diff
@@ -627,16 +639,28 @@ export default Component.extend(FormMixin, {
...
+    customFields =  orderBy(customFields, ['position']);
     return groupBy(requiredFixed.concat(customFields), field => field.get('form'));
   }),
 
   genders       : orderBy(genders, 'name'),
   ageGroups     : orderBy(ageGroups, 'position'),
   countries     : orderBy(countries, 'name'),
   years         : orderBy(years, 'year'),
-  languageForms : orderBy(languageForms, 'item'),
+  languageForms : orderBy(languageForms, 'name'),
```
Comment (norbusan) [inline app/components/forms/orders/order-form.js:663]: Is this correct? I thought the language forms have `item`?
Also, this is an unrelated change, or is it necessary for the current feature?

File: app/components/forms/orders/order-form.js [hunk 3]
```diff
@@ -469,6 +469,17 @@ export default Component.extend(FormMixin, {
       ]
     };
 
+    const isConsentOfRefundPolicyValidation = {
+      optional : true,
+      rules    : [
+        {
+          type   : 'regExp',
+          value  : compulsoryProtocolValidUrlPattern,
+          prompt : this.l10n.t('Please enter a valid URL.')
```
Comment (norbusan) [inline app/components/forms/orders/order-form.js:478]: That text is incorrect I guess. We are asking for consent, not for URLs.

**Review by norbusan — COMMENTED** [review]
Please check the comments in the code I made

File: app/mixins/custom-form.js (hunk 1 above)
Comment (norbusan) [reply app/mixins/custom-form.js:680]: Resolved in meeting

File: app/components/forms/orders/order-form.js (hunk 3 above)
Comment (norbusan) [reply app/components/forms/orders/order-form.js:478]: I will fix this myself in a minute, but first merge this.

**Review by norbusan — APPROVED** [review]
